File: 2015/Day_02/Ruby/test_part_1.rb
require_relative "part_1"
require 'test/unit'

class TestWrappingPaper < Test::Unit::TestCase
    
    def test_get_dimensions
        assert_equal([2, 3, 4], get_dimensions('2x3x4'))
    end
    
    def test_calculate_box_area
        assert_equal(52, calculate_box_area([4,3,2]))
        assert_equal(42, calculate_box_area([1,1,10]))
    end
    
    def test_calculate_small_area
        assert_equal(6, calculate_small_area([4,3,2]))
        assert_equal(1, calculate_small_area([1,1,10]))
    end
end

File: 2015/Day_02/Python/test_part_1.py
from . import part_1


def test_get_dimensions():
    assert part_1.get_dimensions("2x3x4") == [2, 3, 4]
    assert part_1.get_dimensions("1x1x10") == [1, 1, 10]


def test_determine_areas():
    assert part_1.determine_areas([2, 4, 3]) == (6, 52)
    assert part_1.determine_areas([1, 1, 10]) == (1, 42)


def test_full_area():
    assert part_1.full_area((6, 52)) == 58


def test_add_up_all_boxes():
    box_list = ["2x3x4", "1x1x10"]
    assert part_1.add_up_all_boxes(box_list) == 101
